### PR TITLE
Allow users to report anonymously

### LINKED ISSUE #29: Allow user to report messages anonymously
**As a** Community Member  
**I would like** to report messages that violates community [COC](https://bcneng.org/coc) and/or [Netiquette](https://bcneng.org/netiquette) anonymously 
**So that** my identity could be protected.

### PR DESCRIPTION
**Pull request by urielsalis**
Fixes #29

Note: Untested as candebot doesnt seem to run in my machine, but looks simple enough to work.

### CODE REVIEW

File: bot/interact.go [hunk 1]
```diff
@@ -51,8 +51,14 @@ func interactAPIHandler(botContext cmd.BotContext) http.HandlerFunc {
 		case slack.InteractionTypeDialogSubmission:
 			switch message.CallbackID {
 			case "report_message":
+				var user string
+				if message.Submission["anonymous"] == "anonymous" {
+					user = "anonymous"
+				} else {
+					user = message.User.Name
+				}
 				msg := fmt.Sprintf("<@%s> sent a message report:\n- *Reason*: %s\n- *Feeling Scale*: %s of 5\n%s",
```
Comment (smoya) [inline bot/interact.go:60]: When anonymous, this is gonna look weird since it's adding a link to an account that does not exist in theory: `@anonymous`

Comment (urielsalis) [reply bot/interact.go]: Fixed!

File: bot/interact.go [hunk 2]
```diff
@@ -51,12 +51,21 @@ func interactAPIHandler(botContext cmd.BotContext) http.HandlerFunc {
...
 			switch message.CallbackID {
 			case "report_message":
-				msg := fmt.Sprintf("<@%s> sent a message report:\n- *Reason*: %s\n- *Feeling Scale*: %s of 5\n%s",
-					message.User.Name,
-					message.Submission["reason"],
-					message.Submission["scale"],
-					message.State,
-				)
+				var msg string
+				if message.Submission["anonymous"] == "anonymous" {
```
Comment (smoya) [inline bot/interact.go:55]: You could simplify all this code by just making the user, including the `<@name>` part of a var right?
E.g. 

```go
if message.Submission["anonymous"] == "anonymous" {
  user = "anonymous"
} else {
  user = fmt.Sprintf("<@%s>", message.User.Name)
}
```

Comment (urielsalis) [reply bot/interact.go:55]: You are right, done

**Review by smoya — APPROVED** [review]
LGTM! 🚀 🌔

**Pull request merged**